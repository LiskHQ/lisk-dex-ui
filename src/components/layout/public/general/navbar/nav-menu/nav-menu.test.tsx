import React from "react";
import NavMenu from "./nav-menu";
import { render } from "@testing-library/react";

const useRouter = jest.spyOn(require("next/router"), "useRouter");

function renderComponent(props: { isVerticalMenu?: boolean }) {
  return render(<NavMenu isVerticalMenu={props.isVerticalMenu} />);
}

export function mockNextUseRouter(props: {
  route: string;
  pathname: string;
  query: string;
  asPath: string;
}) {
  useRouter.mockImplementationOnce(() => ({
    route: props.route,
    pathname: props.pathname,
    query: props.query,
    asPath: props.asPath,
  }));
}

describe("General layout", () => {
  mockNextUseRouter({
    route: "/mosques/search/[...params]",
    pathname: "/mosques/search/[...params]",
    query: JSON.stringify({
      params: ["New York", "40.712776,-74.005974", "40.712776,-74.005974"],
    }),
    asPath: `/mosques/search/${encodeURIComponent(
      "New York"
    )}/40.712776,-74.005974/40.712776,-74.005974`,
  });

  it("renders correctly with vertical menu", () => {
    const { container } = renderComponent({ isVerticalMenu: true });
    expect(container).toMatchSnapshot();
  });

  it("renders correctly without vertical menu", () => {
    const { container } = renderComponent({ isVerticalMenu: false });
    expect(container).toMatchSnapshot();
  });
});
