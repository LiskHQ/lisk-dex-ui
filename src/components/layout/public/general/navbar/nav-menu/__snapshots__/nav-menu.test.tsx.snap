// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General layout renders correctly with vertical menu 1`] = `
.emotion-0 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-1.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-1.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-1.Mui-selected.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.2);
}

.emotion-1.Mui-disabled {
  opacity: 0.38;
}

.emotion-2 {
  margin: 0;
  color: #1976d2;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-decoration-color: rgba(25, 118, 210, 0.4);
}

.emotion-2:hover {
  text-decoration-color: inherit;
}

<div>
  <ul
    class="MuiList-root MuiList-padding undefined makeStyles-vertical-1 emotion-0"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding makeStyles-item-3 emotion-1"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways makeStyles-link-7 emotion-2"
      >
        <a
          class="makeStyles-link-7 makeStyles-verticalMenu-9"
        >
          About
        </a>
      </a>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding makeStyles-item-3 emotion-1"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways makeStyles-link-7 emotion-2"
      >
        <a
          class="makeStyles-link-7 makeStyles-verticalMenu-9"
        >
          Help & Support
        </a>
      </a>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding makeStyles-item-3 emotion-1"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways makeStyles-link-7 emotion-2"
      >
        <a
          class="makeStyles-link-7 makeStyles-verticalMenu-9"
        >
          Legal & Policy
        </a>
      </a>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding makeStyles-item-3 emotion-1"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways makeStyles-link-7 emotion-2"
      >
        <a
          class="makeStyles-link-7 makeStyles-verticalMenu-9"
        >
          Wallet
        </a>
      </a>
    </li>
  </ul>
</div>
`;

exports[`General layout renders correctly without vertical menu 1`] = `
.emotion-0 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-1.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-1.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-1.Mui-selected.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.2);
}

.emotion-1.Mui-disabled {
  opacity: 0.38;
}

.emotion-2 {
  margin: 0;
  color: #1976d2;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-decoration-color: rgba(25, 118, 210, 0.4);
}

.emotion-2:hover {
  text-decoration-color: inherit;
}

<div>
  <ul
    class="MuiList-root MuiList-padding undefined makeStyles-horizontal-13 emotion-0"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding makeStyles-item-14 emotion-1"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways makeStyles-link-18 emotion-2"
      >
        <a
          class="makeStyles-link-18"
        >
          About
        </a>
      </a>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding makeStyles-item-14 emotion-1"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways makeStyles-link-18 emotion-2"
      >
        <a
          class="makeStyles-link-18"
        >
          Help & Support
        </a>
      </a>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding makeStyles-item-14 emotion-1"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways makeStyles-link-18 emotion-2"
      >
        <a
          class="makeStyles-link-18"
        >
          Legal & Policy
        </a>
      </a>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding makeStyles-item-14 emotion-1"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways makeStyles-link-18 emotion-2"
      >
        <a
          class="makeStyles-link-18"
        >
          Wallet
        </a>
      </a>
    </li>
  </ul>
</div>
`;
